moz_security/sandboxes/heka/analysis/moz_security_cloudtrail.lua

```diff
@@ -86,15 +86,16 @@ end
 
 function genpayload()
     local msg = decode_message(read_message("raw"))
-    local p = {}
+    local p = ""
+    table.sort(msg.Fields, function(a,b) return a.name < b.name end)
     for _, kvpair in ipairs(msg.Fields) do
         if #kvpair.value == 1 then
-            p[kvpair.name] = kvpair.value[1]
+            p = p .. string.format("%s: %s\n", kvpair.name, kvpair.value[1])
         else
-            p[kvpair.name] = kvpair.value
+            p = p .. string.format("%s: %s\n", kvpair.name, kvpair.value)
```

Comment:
You would have to table.concat value otherwise you will get the table address. Also bool is a special case (it cannot be concatted).

Reply:
:+1: - After adding some further tests to see how arrays and bools are handled, I also realized that this else statement is unused when using the `table_to_fields` decoder.